src/analysis/liveness-transfer-function.h

```diff
@@ -47,38 +47,35 @@ class LivenessTransferFunction : public Visitor<LivenessTransferFunction> {
   // run before earlier CFG blocks. This improves performance by
   // reducing the number of state propagations needed, since we are
   // naturally following the backward flow at the beginning.
-  void enqueueWorklist(
-    const std::vector<BlockState<FinitePowersetLattice>>& stateBlocks,
-    std::queue<Index>& worklist) {
-    for (auto it = stateBlocks.rbegin(); it != stateBlocks.rend(); ++it) {
-      worklist.push((*it).getCFGBlock()->getIndex());
+  void enqueueWorklist(CFG* cfg, std::queue<Index>& worklist) {
+    for (auto it = cfg->rbegin(); it != cfg->rend(); ++it) {
+      worklist.push((*it).getIndex());
     }
   }
 
-  // Predecessors depend on use for information.
-  using iterator =
-    std::vector<BlockState<FinitePowersetLattice>*>::const_iterator;
-  iterator depsBegin(BlockState<FinitePowersetLattice>& currBlock) {
-    return currBlock.predecessorsBegin();
-  }
-  iterator depsEnd(BlockState<FinitePowersetLattice>& currBlock) {
-    return currBlock.predecessorsEnd();
+  // Predecessors depend on current basic block for information.
+  _indirect_ptr_vec<BasicBlock> getDependents(const BasicBlock* currBlock) {
```

Comment:
`_indirect_ptr_vec` is supposed to be a private implementation detail, so can we use something like `using iterator BasicBlock::Predecessors::iterator` like we had before?

Reply:
@tlively `getDependents` must be able to return both predecessor, successor, or some arbitrary iterator though.

Reply:
Do we have to require it to have some particular type, or can we just let each `TransferFunction` return whatever iterator types it wants, though?

Reply:
As discussed offline, we will assert that the method exists with the correct parameter types, but for now we won't be checking the return type, since we don't know right now how to assert on all iterables without a C++20 concept.